### PR TITLE
feat(catalog): Add service catalog as a component following new ADR

### PR DESCRIPTION
**Pull request by SamoKopecky**
Part of [operate-first/service-catalog#55](https://github.com/operate-first/service-catalog/issues/55)

~~Blocked by: [operate-first/service-catalog#78](https://github.com/operate-first/service-catalog/pull/78)~~

Create a fully defined entity following the new [ADR 0022](https://github.com/operate-first/blueprint/pull/109), which includes adding:
- `backstage`, `grafana`, `argoCD` APIs
- `smaug`, `s3-bucket`, `crunchy-postgres-db` resources
- `backstage` component
- `argoCD`, `grafana`, `service-catalog` systems
- `SRE` domain

Remove all of the old entity definition files that do not follow the new ADR.

Some of the additions are shared with #2221 since it has been created before this PR was merged.

### CODE REVIEW

**Comment by SamoKopecky** [comment]
No longer blocked.
/unhold

File: service-catalog/components/backstage.yaml
Comment (tumido) [inline service-catalog/components/backstage.yaml:33]: Does it really depend on Argo CD? If Argo CD crashes, does backstage stop working?

```suggestion
    - component:argocd
```

File: service-catalog/components/backstage.yaml
Comment (tumido) [inline service-catalog/components/backstage.yaml:34]: Same as above

File: service-catalog/components/backstage.yaml
Comment (tumido) [inline service-catalog/components/backstage.yaml:38]:
```suggestion
    - argocd
```

**Review by tumido — CHANGES_REQUESTED** [review]
I think there are some structural issues with the "currently available" components in the catalog which pollute the "new and better" structure... Rather remove things like ODF or Observatorium if we can't declare them properly...

Also there should be no operate-first-cloud System entity to catch all Components without a System.

File: service-catalog/systems/service-catalog.yaml
Comment (tumido) [inline service-catalog/systems/service-catalog.yaml:2]: This system is missing postgres crunchy operator Component

**Review by tumido — CHANGES_REQUESTED** [review]

File: service-catalog/components/argo.yaml
Comment (SamoKopecky) [reply service-catalog/components/argo.yaml:1]: Resolved in the latest commit.

File: service-catalog/apis/argocd.yaml
Comment (SamoKopecky) [reply service-catalog/apis/argocd.yaml:4]: Resolved in the latest commit.

File: service-catalog/components/argo.yaml
Comment (SamoKopecky) [reply service-catalog/components/argo.yaml:23]: Resolved in the latest commit.

File: service-catalog/systems/operate-first-cloud.yaml
Comment (SamoKopecky) [reply service-catalog/systems/operate-first-cloud.yaml:4]: Resolved in the latest commit.

File: service-catalog/systems/argocd.yaml
Comment (SamoKopecky) [reply service-catalog/systems/argocd.yaml:4]: Resolved in the latest commit.

File: service-catalog/components/observatorium.yaml
Comment (SamoKopecky) [reply service-catalog/components/observatorium.yaml:5]: Resolved in the latest commit.

File: service-catalog/components/odf.yaml
Comment (SamoKopecky) [reply service-catalog/components/odf.yaml:4]: Resolved a newly pushed commit.

File: service-catalog/components/odh.yaml
Comment (SamoKopecky) [reply service-catalog/components/odh.yaml:4]: Resolved a newly pushed commit.

File: service-catalog/components/dex.yaml
Comment (SamoKopecky) [reply service-catalog/components/dex.yaml:18]: Resolved a newly pushed commit.

**Comment by SamoKopecky** [comment]
About the `operate-first-cloud` system, I let it stay because I thought we would slowly phase it out as new systems come in. However I agree with you that it pollutes the current setup so I'll remove all of the old entities.

File: service-catalog/components/backstage.yaml
Comment (SamoKopecky) [reply service-catalog/components/backstage.yaml:33]: Resolved in the latest commit.

File: service-catalog/components/backstage.yaml
Comment (SamoKopecky) [reply service-catalog/components/backstage.yaml:34]: Resolved in the latest commit.

File: service-catalog/components/backstage.yaml
Comment (SamoKopecky) [reply service-catalog/components/backstage.yaml:38]: Resolved in the latest commit.

File: service-catalog/apis/argocd.yaml
Comment (SamoKopecky) [reply service-catalog/apis/argocd.yaml:12]: This didn't work when I tried it, got the following error: 
```
catalog warn Processor BuiltinKindsEntityProcessor threw an error while validating the entity api:default/argocd; caused by TypeError: /spec/definition must be string - type: string type=plugin entity=api:default/argocd
```

File: service-catalog/apis/grafana.yaml
Comment (SamoKopecky) [reply service-catalog/apis/grafana.yaml:12]: It worked even like this, and I think it may be using the GitHub OAuth login to get the data. But I agree with you that it should be `raw` link, however this needs a new addition to the allowed hosts in the `app-config.yaml`.

File: service-catalog/systems/service-catalog.yaml
Comment (SamoKopecky) [reply service-catalog/systems/service-catalog.yaml:2]: Resolved in the latest commit.

**Comment by SamoKopecky** [comment]
Blocked by https://github.com/operate-first/service-catalog/pull/89
/hold

**Comment by SamoKopecky** [comment]
No longer blocked we just need to remeber to build a new image after merging this to reflect the changes in https://github.com/operate-first/service-catalog/pull/89.

/unhold

**Comment by SamoKopecky** [comment]
All feedback should be addressed. Ready for review.

**Comment by mimotej** [comment]
@SamoKopecky Could you please try to rebase this PR?

**Comment by SamoKopecky** [comment]
> @SamoKopecky Could you please try to rebase this PR? Looks to be similliar issue [here](https://github.com/operate-first/apps/pull/2296#issuecomment-1227057708) @tumido should we capture this as an issue or is this only problem in older PRs, where rebase can solve it?

Should be rabased to `master` now.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
